core/device.h

```diff
@@ -59,11 +66,26 @@ struct device {
 	uint32_t deviceId;	// Always the string hash of the serialNumber
 };
 
+struct fingerprint_record {
+	bool operator<(const fingerprint_record &a) const;
+	uint32_t       model;     // model and libdivecomputer serial number to
+	uint32_t       serial;    //    look up the fingerprint
+	unsigned char *raw_data;  // fingerprint data as provided by libdivecomputer
+	unsigned int   fsize;     // size of raw fingerprint data
```

Comment:
I'm not sure that `unsigned int` is the best data type for expressing sizes. One of the most annoying features of C++ (to me) is `size_t` being unsigned. You are consistent in that choice however, so probably best to keep it.

Reply:
Well, sizes are kinda non-negative... so why do you dislike that?

Reply:
I dislike it, because unsigned integer arithmetic is hard to get right. Things like
```
        if (len1 - 2 < len2)
```
fail if `len1` happens to be 0 or 1, etc. Also once one uses `size_t` for lengths one logically uses `size_t` for indexes and then loops like
```
        for (size_t index = len - 1; index > 0; --index)
```
fail to work.

Moreover, one has to use awkward values such as `std::string::npos` or `((size_t)-1)` to indicate "invalid" or "out of bound".

Of course I understand that this comes from a time where 32-bit was the norm and people wanted to use vectors with more than 2^31 elements. However, now that 64-bit are the norm, it is just painful.

In my own projects, I consistently use `size_t` for lengths and indexes, since this is what the standard library does, but it is kind of annoying...